server/src/main/java/io/crate/planner/GCDanglingArtifactsPlan.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexAction;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.metadata.RelationMetadata;
+
+import com.carrotsearch.hppc.cursors.ObjectCursor;
+
+import io.crate.data.Row;
+import io.crate.data.RowConsumer;
+import io.crate.execution.support.OneRowActionListener;
+import io.crate.planner.operators.SubQueryResults;
+
+public final class GCDanglingArtifactsPlan implements Plan {
+
+    @Override
+    public StatementType type() {
+        return StatementType.MANAGEMENT;
+    }
+
+    @Override
+    public void executeOrFail(DependencyCarrier dependencies,
+                              PlannerContext plannerContext,
+                              RowConsumer consumer,
+                              Row params,
+                              SubQueryResults subQueryResults) {
+        var listener = OneRowActionListener.oneIfAcknowledged(consumer);
+        try {
+            ClusterState clusterState = dependencies.clusterService().state();
```

Comment:
Would be more save to run this inside a cluster state operation on the master node by e.g. using a concrete `RemoveDanglingArtifactsAction/Request`. Otherwise it may process an old state. Any reason not to do this instead?

Reply:
Nope, that should be the correct thing to do.